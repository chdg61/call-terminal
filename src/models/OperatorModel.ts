import { Expose } from 'class-transformer';

export interface IOperatorModel {
    id: number;
    slug: string;
    name: string;
}

export class OperatorModel implements IOperatorModel {

    @Expose()
    public id: number = 0;

    @Expose()
    public slug: string = '';

    @Expose()
    public name: string = '';
}
