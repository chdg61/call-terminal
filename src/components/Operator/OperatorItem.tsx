import * as React from 'react';
import { Link } from 'react-router-dom';
import { IOperatorModel } from '../../models/OperatorModel';

interface Props {
    operator: IOperatorModel;
}

export class OperatorItem extends React.Component<Props> {

    componentDidMount(): void {
    }

    render(): React.ReactElement<any, string | React.JSXElementConstructor<any>> | string | number | {} | React.ReactNodeArray | React.ReactPortal | boolean | null | undefined {
        const { operator } = this.props;
        return (
            <div className="media text-muted pt-3">
                <div className="media-body pb-3 mb-0 small lh-125 border-bottom border-gray">
                    <div className="d-flex justify-content-between align-items-center w-100">
                        <strong className="text-gray-dark">
                            {operator.name}
                        </strong>
                        <Link to={`/refill/${operator.slug}`}>Пополнить</Link>
                    </div>
                </div>
            </div>
        );
    }
}
