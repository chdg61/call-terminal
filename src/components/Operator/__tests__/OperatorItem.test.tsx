import * as React from 'react';
import { shallow } from 'enzyme';
import { OperatorModel } from '../../../models/OperatorModel';
import { OperatorItem } from '../OperatorItem';

jest.mock('react-router-dom', () => ({
    Link: (props) => (
        <a href="#">{props.children}</a>
    ),
}));



describe('OperatorItem', () => {

    it('success operator', () => {
        const operator = new OperatorModel()
        operator.id = 1;
        operator.slug = 'test';
        operator.name = 'Test name';
        const wrapper = shallow(
            <OperatorItem operator={operator} />
        );
        const instance = wrapper.instance();

        expect(wrapper.find('strong.text-gray-dark').text()).toBe('Test name');
        expect(wrapper.html()).toMatchSnapshot();
        expect(wrapper.find('Link').props()).toEqual({
            to: '/refill/test',
            children: 'Пополнить'
        });
    });
});
