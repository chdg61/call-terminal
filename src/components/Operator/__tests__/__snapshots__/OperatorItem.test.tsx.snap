// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OperatorItem success operator 1`] = `"<div class=\\"media text-muted pt-3\\"><div class=\\"media-body pb-3 mb-0 small lh-125 border-bottom border-gray\\"><div class=\\"d-flex justify-content-between align-items-center w-100\\"><strong class=\\"text-gray-dark\\">Test name</strong><a href=\\"#\\">Пополнить</a></div></div></div>"`;
